modules/web-console/src/main/java/org/apache/ignite/console/WebConsoleServer.java

```diff
@@ -129,28 +141,79 @@ public WebConsoleServer(
 
         registerEventBusConsumers();
 
+        boolean ssl = !F.isEmpty(cfg.getKeyStore()) || !F.isEmpty(cfg.getTrustStore());
+
+        int port = cfg.getPort();
+
+        // Add redirect to HTTPS.
+        if (ssl & port != 80) {
+            vertx
+                .createHttpServer()
+                .requestHandler(req -> {
+                    String origin = Utils.origin(req).replace("http:", "https:");
+
+                    if (port != 443)
+                        origin += ":" + port;
+
+                    req.response()
+                        .setStatusCode(HTTP_MOVED_PERM)
+                        .setStatusMessage("Server requires HTTPS")
+                        .putHeader(HttpHeaders.LOCATION, origin)
+                        .end();
+                })
+                .listen(80);
+        }
+
         Router router = Router.router(vertx);
 
         router.route().handler(CookieHandler.create());
         router.route().handler(BodyHandler.create());
         router.route().handler(SessionHandler.create(LocalSessionStore.create(vertx)));
         router.route().handler(UserSessionHandler.create(auth));
 
+        if (!F.isEmpty(cfg.getWebRoot()))
+            router.route().handler(StaticHandler.create(cfg.getWebRoot()));
+
         router.route("/eventbus/*").handler(sockJsHnd);
 
         registerRestRoutes(router);
 
         registerVisorTasks();
 
-        // Start HTTP server for browsers and web agents.
         HttpServerOptions httpOpts = new HttpServerOptions()
             .setCompressionSupported(true)
             .setPerMessageWebsocketCompressionSupported(true);
 
+        if (ssl) {
+            httpOpts.setSsl(true);
+
+            JksOptions jks = Utils.jksOptions(cfg.getKeyStore(), cfg.getKeyStorePassword());
```

Comment:
Why don't you use static import for `Utils.jksOptions`?

Reply:
OK.